File: training_camps/trainingcamp2022/2/contest/l2.py
input()
nums=map(int,input().split())

def istprime(n):
	sqr=n**(1/2)
	print(sqr)
	sqr3=n**(1/3)
	if n%sqr==0: return True
	#elif:
	else: return False

for i in nums:
	if i<3:
		print("NO")		
	elif istprime(i):
		print("YES")
	else :
		print("NO")

File: training_camps/trainingcamp2022/2/contest/L2.cpp
#include <bits/stdc++.h>
using namespace std;
bool istprime(int n){
  //float root=sqrt(n);
  int cast=(int)(root);
  cout<<cast<<endl;
  cout<<root<<endl;
    if(n%cast==0){
       return true;
    }else{
     return false;
   }
}
int main(){
  int times,num;
  cin>>times;
  int nums[times];
  for(int i=0;i<times;i++){
   cin>>num;
   if(istprime(num)){
      cout<<"YES"<<endl;
    }
    else{
      cout<<"NO"<<endl;
    }
  }
  return 0;
}
